### ISSUE TITLE
Terraform console crashes with local variables

### ISSUE THREAD

**Issue by YektaLeblebici** [hashicorp/terraform#18413]
I was simply playing with the "terraform console", and noticed that if I ever write anything starting with "local." to the REPL, console crashes.

### Terraform Version
```
Terraform v0.11.7
```

### Terraform Configuration Files
It was an empty dir.

### Crash Output
https://gist.github.com/YektaLeblebici/d4b88d7df90f68e3f75380a0643a103c

### Expected Behavior
Maybe throw some kind of error?

### Actual Behavior
It crashed.

### Steps to Reproduce
1. Run `terraform console`
2. Write `local.foo' to the REPL and press enter.

**Comment by apparentlymart** [comment]
Hi @AranelSurion! Sorry for this crash and thanks for reporting it.

The relevant portion of Terraform has been changed significantly in the development branch for [the next major release](https://www.hashicorp.com/blog/terraform-0-1-2-preview), so I've labelled this so we'll find it while we're doing our testing pass for the release and we can see if this issue is still present in the new implementation, and if we need to do anything else here to make this work as expected.

An important thing we changed in the dev branch was having Terraform run a graph walk to populate variables, locals, and outputs before displaying the console prompt, which I expect is at least a partial solution to this problem.

**Comment by apparentlymart** [comment]
Hi again @YektaLeblebici!

The changes I mentioned previously are now merged into master ready to be included in the forthcoming v0.12.0 release, so I tested this in the v0.12.0-alpha2 prerelease build with the following simple configuration:

```hcl
locals {
  foo = "hello"
}
```

```
$ terraform console
> local.foo
hello
> local.bar

Error: Reference to undeclared local value

  on <console-input> line 1:
  (source code not available)

A local value with the name "bar" has not been declared.

> 
```

The console's handling of errors within entered expressions is still not perfect, but this does illustrate that referring to defined locals is working and referring to undefined locals is returning a proper error rather than a crash.

Since this fix is in master already, I'm going to close this out. Thanks again for reporting this!
